src/js/Controllers/RpcFactory.js

```diff
@@ -429,16 +430,38 @@ class RpcFactory {
         return msg  
     }
     static SetDisplayLayoutResponse(rpc) {
-        return ({
-            "jsonrpc": "2.0",
-            "id": rpc.id,
-            "result": {
-                "method": rpc.method,
-                "code": 0,
-                "displayCapabilities": capabilities.displayCapabilities,
-                "softButtonCapabilities": capabilities.softButtonCapabilities
+        var layout = rpc.params.displayLayout;
+        var supportedTemplates = ["DEFAULT", "MEDIA", "NON-MEDIA", "LARGE_GRAPHIC_ONLY", 
+        "LARGE_GRAPHIC_WITH_SOFTBUTTONS", "GRAPHIC_WITH_TEXTBUTTONS", "TEXTBUTTONS_WITH_GRAPHIC", 
+        "TEXTBUTTONS_ONLY", "TILES_ONLY", "TEXT_WITH_GRAPHIC", "GRAPHIC_WITH_TEXT", "DOUBLE_GRAPHIC_WITH_SOFTBUTTONS"];
+        if (supportedTemplates.includes(layout)) {
+            if (layout == "DEFAULT") {
+                layout = "MEDIA"
             }
-        })        
+            return ({
+                "jsonrpc": "2.0",
+                "id": rpc.id,
+                "result": {
+                    "method": rpc.method,
+                    "code": 0,
+                    "displayCapabilities": capabilities[layout].displayCapabilities ? capabilities[layout].displayCapabilities : null,
+                    "softButtonCapabilities": capabilities[layout].softButtonCapabilities ? capabilities[layout].softButtonCapabilities : null,
+                    "buttonCapabilities": capabilities[layout].buttonCapabilities ? capabilities[layout].buttonCapabilities : null
```

Comment:
Core differentiates between NULL and an omitted value, as such, it considers this invalid. These values should be omitted instead.